### PR TITLE
Document event types in notification docs

### PR DESCRIPTION
**Pull request by poornas**
## Motivation and Context
Document event types missing that are supported.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

File: docs/bucket/notifications/README.md [hunk 1]
```diff
@@ -5,8 +5,10 @@ Events occurring on objects in a bucket can be monitored using bucket event noti
 | Supported Event Types   |                                            |                          |
 | :---------------------- | ------------------------------------------ | ------------------------ |
 | `s3:ObjectCreated:Put`  | `s3:ObjectCreated:CompleteMultipartUpload` | `s3:ObjectAccessed:Head` |
-| `s3:ObjectCreated:Post` | `s3:ObjectRemoved:Delete`                  |                          |
-| `s3:ObjectCreated:Copy` | `s3:ObjectAccessed:Get`                    |                          |
+| `s3:ObjectCreated:Post` | `s3:ObjectRemoved:Delete`                  |  `s3:ObjectRemoved:DeleteMarkerCreated`                        |
+| `s3:ObjectCreated:Copy` | `s3:ObjectAccessed:Get`                    | `s3:ObjectAccessed:GetLegalHold`                         |
+| `s3:ObjectCreated:PutRetention` | `s3:ObjectAccessed:Get`                    | `s3:ObjectCreated:PutLegalHold`                         |`s3:ObjectAccessed:GetRetention`|
+| `s3:Replication:OperationFailedReplication` | `s3:BucketCreated`                    | `s3:BucketRemoved`|
```
Comment (harshavardhana) [inline docs/bucket/notifications/README.md]: Looks like some markdown formatting issue here can you fix this on your end?

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Looks like markdown tabbing/formatting is off - you may apply the following diff. 

```diff
diff --git a/docs/bucket/notifications/README.md b/docs/bucket/notifications/README.md
index f61ff08c4..c91424aea 100644
--- a/docs/bucket/notifications/README.md
+++ b/docs/bucket/notifications/README.md
@@ -2,11 +2,13 @@
 
 Events occurring on objects in a bucket can be monitored using bucket event notifications. Event types supported by MinIO server are
 
-| Supported Event Types   |                                            |                          |
-| :---------------------- | ------------------------------------------ | ------------------------ |
-| `s3:ObjectCreated:Put`  | `s3:ObjectCreated:CompleteMultipartUpload` | `s3:ObjectAccessed:Head` |
-| `s3:ObjectCreated:Post` | `s3:ObjectRemoved:Delete`                  |                          |
-| `s3:ObjectCreated:Copy` | `s3:ObjectAccessed:Get`                    |                          |
+| Supported Event Types                       |                                            |                                        |                                  |
+| :----------------------                     | ------------------------------------------ | ------------------------               |                                  |
+| `s3:ObjectCreated:Put`                      | `s3:ObjectCreated:CompleteMultipartUpload` | `s3:ObjectAccessed:Head`               |                                  |
+| `s3:ObjectCreated:Post`                     | `s3:ObjectRemoved:Delete`                  | `s3:ObjectRemoved:DeleteMarkerCreated` |                                  |
+| `s3:ObjectCreated:Copy`                     | `s3:ObjectAccessed:Get`                    | `s3:ObjectAccessed:GetLegalHold`       |                                  |
+| `s3:ObjectCreated:PutRetention`             | `s3:ObjectAccessed:Get`                    | `s3:ObjectCreated:PutLegalHold`        | `s3:ObjectAccessed:GetRetention` |
+| `s3:Replication:OperationFailedReplication` | `s3:BucketCreated`                         | `s3:BucketRemoved`                     |                                  |
 
 Use client tools like `mc` to set and listen for event notifications using the [`event` sub-command](https://docs.min.io/docs/minio-client-complete-guide#events). MinIO SDK's [`BucketNotification` APIs](https://docs.min.io/docs/golang-client-api-reference#SetBucketNotification) can also be used. The notification message MinIO sends to publish an event is a JSON message with the following [structure](https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html).
 
```

**Review by kannappanr — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#2670b3b7-897e-4bef-b680-c551508b7d40) |
<a name="2670b3b7-897e-4bef-b680-c551508b7d40"></a>
### 10636-98fb06a/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:32604
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 38cb75dc827a:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 8 seconds
(4/15) Running aws-sdk-ruby tests ... done in 20 seconds
(5/15) Running awscli tests ... done in 2 minutes and 47 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 3 minutes and 52 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 39 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 45 seconds
{
  "args": {
    "bucketName": "minio-go-test-94szfa2f6lvg3lmw",
    "objectName": "test-object",
    "opts": "",
    "size": -1
  },
  "duration": 1025,
  "function": "PutObject(bucketName, objectName, reader,size,opts)",
  "message": "Unexpected size",
  "name": "minio-go: testPutObjectStreaming",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 11 minutes and 29 seconds
(10/15) Running minio-js tests ... FAILED in 45 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 80,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:682:10) at endReadableNT (_stream_readable.js:1252:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(10/15) Running minio-py tests ... done in 19 minutes and 16 seconds
(11/15) Running s3cmd tests ... done in 2 minutes and 13 seconds
(12/15) Running s3select tests ... done in 38 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10636-98fb06a

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#d746dc2c-4cdd-40c3-9f81-9d7870770923) |
<a name="d746dc2c-4cdd-40c3-9f81-9d7870770923"></a>
### 10636-76bb3bf/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:32546
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5acb52a55d1c:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 8 seconds
(4/15) Running aws-sdk-ruby tests ... done in 29 seconds
(5/15) Running awscli tests ... done in 3 minutes and 0 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 3 minutes and 29 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 36 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 48 seconds
{
  "args": {
    "bucketName": "minio-go-test-3zql4o6q1ghbdij9",
    "objectName": "test-object",
    "opts": "",
    "size": -1
  },
  "duration": 1098,
  "function": "PutObject(bucketName, objectName, reader,size,opts)",
  "message": "Unexpected size",
  "name": "minio-go: testPutObjectStreaming",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... FAILED in 4 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[multi-part upload]",
  "duration": 906,
  "status": "FAIL",
  "error": "java.net.SocketException: Connection reset >>> [java.base/java.net.SocketInputStream.read(SocketInputStream.java:186), java.base/java.net.SocketInputStream.read(SocketInputStream.java:140), okio.Okio$2.read(Okio.java:140), okio.AsyncTimeout$2.read(AsyncTimeout.java:237), okio.RealBufferedSource.indexOf(RealBufferedSource.java:358), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:230), okhttp3.internal.http1.Http1ExchangeCodec.readHeaderLine(Http1ExchangeCodec.java:242), okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.java:213), okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.java:115), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:94), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:43), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:88), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:229), okhttp3.RealCall.execute(RealCall.java:81), io.minio.MinioClient.execute(MinioClient.java:1014), io.minio.MinioClient.uploadPart(MinioClient.java:7781), io.minio.MinioClient.putObject(MinioClient.java:4706), io.minio.MinioClient.putObject(MinioClient.java:4881), FunctionalTest.testPutObject(FunctionalTest.java:740), FunctionalTest.putObject_test(FunctionalTest.java:800), FunctionalTest.runTests(FunctionalTest.java:3791), FunctionalTest.main(FunctionalTest.java:4058)]"
}
(9/15) Running minio-js tests ... FAILED in 45 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 76,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:682:10) at endReadableNT (_stream_readable.js:1252:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(9/15) Running minio-py tests ... done in 19 minutes and 7 seconds
(10/15) Running s3cmd tests ... done in 2 minutes and 20 seconds
(11/15) Running s3select tests ... done in 38 seconds
(12/15) Running security tests ... done in 0 seconds

Executed 12 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10636-76bb3bf

**Pull request merged**